src/main/java/seedu/address/scraper/ScraperManager.java

```diff
@@ -96,19 +97,28 @@ public void startScraping() throws IOException {
             return;
         }
 
-        List<Mission> missions = getMissions();
-        List<Quest> quests = getQuests();
-        List<Student> students = new ArrayList<>();
+        // Run JavaFX-modifying functions after JavaFX Thread is done.
+        Platform.runLater(() -> {
```

Comment:
Good idea to include threads for GUI!

Reply:
Agreed! Really helps to speed up the loading! :)